extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaTransclusionModel.js

```diff
@@ -2,148 +2,201 @@
+	OO.inheritClass( ve.dm.WikiaTransclusionModel, ve.dm.MWTransclusionModel );
 
-/* Methods */
+	/* Methods */
 
-/**
- * @inheritdoc
- */
-ve.dm.WikiaTransclusionModel.prototype.fetchRequest = function ( titles, specs, queue ) {
-	//if we want to make another request, we should update the requests array as well
-	this.requests.push( this.fetchInfoboxParamsRequest( titles, specs, queue ) );
-
-	return ve.init.target.constructor.static.apiRequest( {
-		action: 'templateparameters',
-		titles: titles.join( '|' )
-	} )
-	.done( this.fetchRequestDone.bind( this, specs ) )
-	.always( this.fetchRequestAlways.bind( this, queue ) );
-};
-
-ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequest = function ( titles, specs, queue ) {
-	return ve.init.target.constructor.static.apiRequest( {
-		action: 'query',
-		prop: 'infobox',
-		titles: titles.join( '|' )
-	} )
-	.done( this.fetchInfoboxParamsRequestDone.bind( this, specs ) )
-	.always( this.fetchRequestAlways.bind( this, queue ) );
-};
-
-/**
- * @inheritdoc
- */
-ve.dm.WikiaTransclusionModel.prototype.fetchRequestDone = function ( specs, data ) {
-	var page, i, id;
-	if ( data && data.pages ) {
-		for ( id in data.pages ) {
-			page = data.pages[id];
-			if ( !specs[page.title] ) {
-				specs[page.title] = {
-					title: data.pages[id].title,
-					description: '',
-					params: {},
-					paramOrder: page.params
-				};
-			}
-			// Map parameters from flat array to collection where parameter name is the key and value is empty
-			// because later it's extended with ve.dm.MWTemplateSpecModel.prototype.getDefaultParameterSpec
-			for ( i = 0; i < page.params.length; i++ ) {
-				specs[page.title].params[page.params[i]] = {};
+	ve.dm.WikiaTransclusionModel.prototype.pushFetchRequests = function ( titles, specs, queue ) {
+		var templateParamsRequest = this.fetchTemplateParamsRequest( titles, specs, queue ),
+			infoboxParamsRequest = this.fetchInfoboxParamsRequest( titles, specs, queue );
+
+		$.when( templateParamsRequest, infoboxParamsRequest )
+			.then( function( templateParamsResponse, infoboxParamsResponse ) {
+				if ( templateParamsResponse[1] === 'success' ) {
+					this.templateParamsRequestDone( specs, templateParamsResponse[0] );
+				}
+				if ( infoboxParamsResponse[1] === 'success' ) {
+					this.infoboxParamsRequestDone( specs, infoboxParamsResponse[0] );
+				}
+			}.bind( this ) )
+			.always( function ( queue ) {
+				this.process( queue );
+			}.bind( this, queue ) );
+
+		this.requests.push( templateParamsRequest );
+		this.requests.push( infoboxParamsRequest );
+	};
+
+	/**
+	 * @inheritdoc
+	 */
+	ve.dm.WikiaTransclusionModel.prototype.fetchTemplateParamsRequest = function ( titles, specs, queue ) {
+		return ve.init.target.constructor.static.apiRequest( {
+			action: 'templateparameters',
+			titles: titles.join( '|' )
+		} );
+	};
+
+	ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequest = function ( titles, specs, queue ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "specs". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "queue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)